java-checks/src/main/java/org/sonar/java/checks/RedundantTypeCastCheck.java

```diff
@@ -57,14 +58,16 @@ public void visitNode(Tree tree) {
       return;
     }
     TypeCastTree typeCastTree = (TypeCastTree) tree;
-    Type cast = typeCastTree.type().symbolType();
-    Type target = targetType(typeCastTree);
     Type expressionType = typeCastTree.expression().symbolType();
     if (isPrimitiveWrapperInConditional(expressionType, typeCastTree) || requiredForMemberAccess(typeCastTree)) {
       // Primitive wrappers excluded because covered by S2154
       return;
     }
-    if (target != null && (isRedundantNumericalCast(cast, expressionType) || isUnnecessarySubtypeCast(expressionType, typeCastTree, target))) {
+    Type cast = typeCastTree.type().symbolType();
+    Type target = targetType(typeCastTree);
+    boolean isArgument = skipParentheses(typeCastTree.parent()).is(Tree.Kind.ARGUMENTS);
```

Comment:
You only need `isArgument` within method `isUnnecessarySubtypeCast()`, and you can computed it from there, as you are anyway using passing `typeCastTree` as parameter.